maths/gamma_recursive.py

```diff
@@ -0,0 +1,84 @@
+"""
+Gamma function is a very useful tool in physics.
+It helps calculating complex integral in a convenient way.
+for more info: https://en.wikipedia.org/wiki/Gamma_function
+"""
+
+# Importing packages
+from math import sqrt, pi
+from re import match
+from typing import Union
+
+
+def gamma(num : Union[int, float]) -> Union[int, float]:
+    """
+    Calculates the value of Gamma function of num
+    where num is either an integer (1,2,3..) or a half-integer (0.5,1.5,2.5...).
+    Implemented using recursion
+    Examples:
+    >>> Gamma of: 0.5
+    √π
+    >>> Gamma of: 2
+    1
+    >>> Gamma of: 3.5
+    1.875√π
+    """
+    if num == 1:
+        return 1
+    elif num == 0.5:
+        return sqrt(pi)
+    elif num > 1:
+        return (num - 1) * gamma(num - 1)
+    # Error
+    return -2
+
+
+def test_gamma() -> None:
+    """
+    >>> test_gamma()
+    """
+    assert sqrt(pi) == gamma(0.5)
+    assert 1 == gamma(1)
+    assert 1 == gamma(2)
+
+
+if __name__ == "__main__":
+    # Initialize boolean
+    number = True
+    # Get input from user
```

Comment:
```suggestion
    is_number = True
```
The comments do not say more than the code does so drop them.

`is_` makes it clear to the reader that this variable is a Boolean.